### PR TITLE
Redo term removal in main

### PR DESCRIPTION
**Pull request by ChumpChief**
This reapplies #14959, but in `main`.  Two changes as compared to that PR:
1. Restore a "backcompat" fixup in DeltaManager.  Some of our snapshot tests predate "term", and so they contain messages without term stamped on them.  In most cases this wouldn't matter for snapshot tests.  However, in some cases the processed ops themselves make it into the snapshot (e.g. notably the catchupOps for merge-tree).  Someone must have updated the snapshots since term was added (and more importantly, the "backcompat" fixup), so even though the original message in messages.json doesn't have "term", the catchupOps version of that same message does.  This results in a snapshot discrepancy when the fixup is removed.
    * It could be reasonable to just update the snapshots to un-fix-up the ops (this would better match the original behavior), but since I intend a further change to remove term entirely (which will touch the snapshot content again) I'd prefer to defer snapshot content updates until that time.
3. Remove a small technically-breaking-change from container-loader.  This was in a deprecated interface that is no longer exported in `next`, so to play by the rules I'll wait to update the interface in `next`.

### CODE REVIEW

File: packages/loader/container-loader/src/deltaManager.ts [hunk 1]
```diff
@@ -988,7 +981,6 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
 		if (message.term === undefined) {
```
Comment (ChumpChief) [inline packages/loader/container-loader/src/deltaManager.ts:981]: This is the fixup that came back.

File: common/lib/protocol-definitions/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@fluidframework/protocol-definitions",
-	"version": "1.2.0",
+	"version": "2.0.0",
```
Comment (ChumpChief) [inline common/lib/protocol-definitions/package.json:3]: @tylerbutler FYI - I think it's still correct to bump in this PR, and once @ghoshkaj 's changes are in too then they can both go out with 2.0.0.

File: packages/loader/container-loader/src/test/deltaManager.spec.ts [hunk 3]
```diff
@@ -81,7 +81,7 @@ describe("Loader", () => {
 					noopCountFrequency,
 				);
 
-				await deltaManager.attachOpHandler(0, 0, 1, {
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/test/deltaManager.spec.ts:84]: maybe add a var, like `const DefaultTermValue = 1 as const`, and a comment on that value. you could do this once per package and not export it.

**Review by anthony-murphy — APPROVED** [review]
Client looks good. Server looks fine to me, as long as both term and epoch really aren't needed anymore, assume you had someone from server look at the last one.

**Comment by ChumpChief** [comment]
> Client looks good. Server looks fine to me, as long as both term and epoch really aren't needed anymore, assume you had someone from server look at the last one.

Yep, Gary and Pradeep reviewed the last one.

**Pull request merged**